aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -576,27 +582,53 @@ void gemm<at::BFloat16>(CUDABLAS_GEMM_ARGTYPES(at::BFloat16)) {
       ldc,
       CUDA_R_32F,
       CUBLAS_GEMM_DEFAULT_TENSOR_OP));
-  TORCH_CUDABLAS_CHECK(cublasSetMathMode(handle, CUBLAS_DEFAULT_MATH));
 }
-#endif // !defined(USE_ROCM)
+#endif // defined(CUDA_VERSION) && CUDA_VERSION >= 11000
 
-#if !defined(USE_ROCM) && !defined(_MSC_VER)
+#if (defined(USE_ROCM) && ROCM_VERSION >= 50600) || (defined(CUDA_VERSION) && CUDA_VERSION >= 11000 && !defined(_MSC_VER))
 
 namespace {
+
+#if defined(USE_ROCM) && ROCM_VERSION >= 50600
+// hipify maps hipDataType -> hipDataType, but hipblaslt expects hipblasDatatype_t
+#define hipDataType hipblasDatatype_t
+// hipblaslt returns hipblasStatus_t, but hipify of TORCH_CUDABLAS_CHECK expects rocblas_status
+#define WORKAROUND_TORCH_CUDABLAS_CHECK(EXPR)                         \
```

Comment:
A less intrusive way is `#undef` and then re-`#define` TORCH_CUBLAS_CHECK` in this file.

```
#undef TORCH_CUBLAS_CHECK
#define TORCH_CUBLAS_CHECK(EXPR) // The current WORKAROUND_TORCH_CUBLAS_CHECK
``` 

If the new `TORCH_CUBLAS_CHECK` still need to handle two types. Use templates.